augur/filter.py

```diff
@@ -390,10 +461,325 @@ def include_by_query(metadata, include_where):
     >>> include_by_query(metadata, "region=europe")
     {'strain2'}
 
+    If the column referenced in the given query does not exist, skip the filter.
+
+    >>> include_by_query(metadata, "missing_column=value")
+    set()
+
     """
     column, op, value = parse_filter_query(include_where)
-    included = op(metadata[column].astype(str).str.lower(), value.lower())
-    return set(metadata[included].index.values)
+
+    if column in metadata.columns:
+        included_index = op(metadata[column].astype(str).str.lower(), value.lower())
+        included = set(metadata[included_index].index.values)
+    else:
+        # Skip the inclusion filter if the requested column does not exist.
+        included = set()
+
+    return included
+
+
+def construct_filters(args, metadata, sequence_index):
+    """Construct lists of filters and inclusion criteria based on user-provided arguments.
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        Command line arguments provided by the user.
+    metadata : pandas.DataFrame
+        Metadata to be filtered.
```

Comment:
On further reflection, this metadata argument is not necessary except to check the available metadata columns. We can drop this argument and the check for fields in the metadata columns and let either the individual filters or the `apply_filters` function handle missing data.

We still need the sequence index to support sequence-based filters, although we could consider passing around the path of the sequence index file instead of the data frame itself. If the sequence index becomes too large, passing the filename could be a better solution.

Reply:
+1 to this change, particularly

> I would expect an explicit request by the user for a field that doesn't exist to produce a helpful error message instead.